### PR TITLE
Refactor formatting of 'list' command (2/3): reduce code duplication

### PR DESCRIPTION
**Pull request by danieleades**
this diff is stacked on #211

further improvements can be made here yet. There's a lot of duplication between the two list 'variants'.

### CODE REVIEW

**Comment by foresterre** [comment]
Is there a way to show just the changes of this PR? With bitbucket, if you relay the branch of this PR, to point to the branch of the PR it depends upon, then when the PR it depends upon is merged into the main branch, it will then automatically redirect this branch to point to main. 

This would make it easier to review the changes, since I think (so far I only had a quick look) some refactoring makes total sense and improves the readability, but other items maybe a difference in taste of code style :).

**Comment by foresterre** [comment]
Hmm, I guess I could simply look at one commit instead of the whole change set.

**Comment by danieleades** [comment]
> Hmm, I guess I could simply look at one commit instead of the whole change set.

yeah i think that's the way to do it in github. github doesn't really have great support for 'stacked' diffs

**Comment by aewag** [comment]
> Is there a way to show just the changes of this PR?

Just stumbled upon this discussion and had the same issue a month ago:

https://github.com/Fraunhofer-AISEC/archie/pull/13#issuecomment-977601869

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

TL;DR: Github can do this, but only if the dependant branch is in this repo and not in a fork.

**Comment by danieleades** [comment]
closing in favour of #213
